project/app/api/visualization.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from fastapi import APIRouter
+
+from app.utils.security.header_checking import AuthRouteHandler
+from app.api.models import HistogramRequest, LineGraphRequest
+from app.utils.visualizations import histogram, line_graph
+
+# global variables and services
+router = APIRouter(route_class=AuthRouteHandler)
+log = logging.getLogger(__name__)
+
+
+@router.route("/viz/linegraph")
+def return_line_graph(data: LineGraphRequest):
+    """Endpoint produces a line graph of a student's squad scores over time
+
+        Input: A list containing the history of the students scores, in
+        chronological order. And student's name.
+
+        Output: Plotly JSON for web to display using plotly.js on the
+        parent dashboard"""
+    return line_graph.line_graph(data.ScoreHistory, data.StudentName)
```

Comment:
Add a blank line after docstring for [PEP8 compliance](https://www.python.org/dev/peps/pep-0257/)

Reply:
As per PEP-257 - Docstring conventions:
```
There's no blank line either before or after the docstring.
```
unless its a class's main Docstring then:
```
Insert a blank line after all docstrings (one-line or multi-line) that document a class -- 
generally speaking, the class's methods are separated from each other by a single 
blank line, and the docstring needs to be offset from the first method by a blank line.
```

Reply:
This just applies to one-line docstrings (outside of a class's):
``` 
There's no blank line either before or after the docstring.
```

The intro to Multi-line Docstrings says this:
```
Multi-line docstrings consist of a summary line just like a one-line docstring, 
followed by a blank line, followed by a more elaborate description. 
```

Reply:
that is only within the docstring itself, after termination there is no blank line.

Reply:
e.g
```
"""[summary]

[long summary]
Arguments:
"""
```

Reply:
Ah okay, I just misread that (multiple times). Thanks for clarifying!